Editor/CoreAPI/StateManager.cs

```diff
@@ -223,6 +223,7 @@ public string DeploymentBuildFolderPath
 
         public Action OnUserProfileUpdated { get; set; }
         public Action OnFleetChanged { get; set; }
+        public Action OnUserProfileSelected { get; set; }
```

Comment:
We shouldn't be adding a new event that's identical (or close to it) to another event, `OnUserProfileUpdated`. I'm guessing this was done because of the bug that was introduced when `OnUserProfileUpdated` was invoked in SaveProfiles.

Reply:
They are not close in function, while there is some overlap in when they trigger I do not want to reset the compute and call an api every data entry change. `OnUserProfileUpdated` is called whenever we update the data on a user profile, whereas `OnUserProfileSelected` is invoked upon using our dropdown to select a new profile.
`OnUserProfileUpdated` can trigger whilst still on the same userprofile, whereas `OnUserProfileSelected` is the act of changing over from 1 profile to the next.